### ISSUE TITLE
only one instance of babel-polyfill is allowed

### ISSUE THREAD

**Issue by fk-hb** [lancedikson/bowser#231]
Hi,
Using the version '2.0.0-alpha.1', I got the error:
only one instance of babel-polyfill is allowed

We use browserify to bundle our client side js file.
browserify will include multiple times a package if it uses different version (or if not located in the same directory). So if you fix your own dependency on babel-polyfill and we (or some other package) fix another one, we will have this error.

I think babel-polyfill is designed to be referenced only once by the program entry point. Why making it a dependency of bowser?

**Comment by lancedikson** [comment]
Hi @fk-hb. You're right that `babel-polyfill` should be included only once.
The thing is that current alpha version considers `dist/bowser.compiled.js` as the main file and when you require the library using `require('bowser'), you will for sure include the whole bundle of the library with polyfill bundled inside. It's made for users, who don't have any babel processing.

In the case when you have your own babel-polyfill and the whole building process I'd recommend you using bowser from sources: `require('bowser/src/bowser')`. In that case, you'll get plain bowser without polyfills and anything. Maybe, I'll change the `main` file configuration so that it would point on `src/bowser.js` in the stable version. But, let's see how it goes.

Please, let me know if it works for you.

**Comment by fk-hb** [comment]
I used require('bowser/src/bowser') and I no more have the problem.
I think most of users of the library will need the unbundled version... why not make:
require('bowser') and require('bowser/bundled')?

**Comment by lancedikson** [comment]
Yeah, good point, I'll change that in the next alpha release at the beginning of the next week :)

**Comment by fk-hb** [comment]
Hi @lancedikson,
I used the last 2.0.0-alpha.3 and now, I can directly call require('bowser') to use the unbundled version.
But I have an issue with browserify (see https://github.com/babel/babelify/issues/157): you need to instruct browserify that your code needs a babelify transform using the es2015 preset.
You just need to add the following to your package.json to make it work:
`"browserify": {
    "transform": [["babelify", { "presets": ["es2015"] }]]
  }`

**Comment by lancedikson** [comment]
Hi, @fk-hb. Do you mean that we need to add `"browserify": { "transform": [["babelify", { "presets": ["es2015"] }]] }` to `package.json` of bowser? I thought they mentioned this lines of code in the context of the project's package.json, not npm-modules' ones. Maybe it would be better for you to use `bowser/es5`? :)
Anyways, in the end, I guess the `bowser/es5` will be the main one and it will be the file you import making `require('bowser')`, since requiring sources seems a bit messy currently — no one will get profit from requiring sources, neither frontend packages, nor backend (node.js) ones.

**Comment by fk-hb** [comment]
Hi @lancedikson,
thank you for your feedback. If I use require('bowser/es5') It works.
es5.js is already transformed and it does not bundle babel-polyfill. It perfectly fits my needs.
Thank you.
